admin/src/utils/wrapWithErrorHandling.js

```diff
@@ -1,3 +1,12 @@
 export default cb => (effects, ...args) => {
-    return cb(effects, ...args).then(value => value, error => effects.setError(error.message));
+    try {
+        const result = cb(effects, ...args);
```

Comment:
you can wrap cb(effects, ...args) in a `Promise.resolve` and avoid the test for `then`